backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoice/ManualInvoiceService.java

```diff
@@ -235,70 +236,77 @@ private CreateManualInvoiceLineRequest buildManualInvoiceLine(
 			@NonNull final BPartnerLocationAndCaptureId bPartnerLocationAndCaptureId,
 			@NonNull final ZoneId zoneId)
 	{
-		final IPricingResult pricingResult = getPricingResult(request.getHeader(),
-															  requestLine,
-															  priceListId,
-															  countryId,
-															  zoneId);
-
 		final ProductPrice manualPrice = request.getPriceEntered(requestLine).orElse(null);
 
+		TaxId taxId = Optional.ofNullable(requestLine.getVatCodeId())
+				.map(this::getTaxByVatId)
+				.orElse(null);
```

Comment:
```suggestion
		final VatCodeId vatCodeId = requestLine.getVatCodeId();
		TaxId taxId = (vatCodeId != null) ? getTaxByVatId(vatCodeId) : null;
```

not everything that *can* be cone with streams&lambdas *should* be done with streams&lambdas..

Reply:
agree, but here I don't see why one option is better than the other :-?

Reply:
I aggree, this one is OK, but I propose to change it so that we have `if ? then : else` twice in the method.
So that the reader, after having wrapped her/his head around the first  `if ? then : else` doesn't have to take on an `Optional.map.orElse;` expression next.